src/modules/exchangeProvider.js

```diff
@@ -60,16 +61,22 @@ const onExchangeContractChange = (newVal, oldVal, objectPath) => {
     if (newVal && store.getState().augmintToken.isConnected) {
         console.debug("exchangeProvider - new Exchange contract. Dispatching refreshOrders()");
         store.dispatch(refreshOrders());
+
+        const userAccount = store.getState().web3Connect.userAccount;
+        const augmintToken = store.getState().augmintToken;
+
+        store.dispatch(fetchTrades(userAccount, augmintToken.contract.deployedAtBlock, "latest"));
         setupListeners();
     }
 };
 
 // event NewOrder(uint indexed orderId, address indexed maker, uint price, uint tokenAmount, uint weiAmount);
-const onNewOrder = (orderId, maker, price, tokenAmount, weiAmount) => {
+const onNewOrder = function(orderId, maker, price, tokenAmount, weiAmount) {
     console.debug("exchangeProvider.onNewOrder: dispatching refreshExchange() and refreshOrders()");
     store.dispatch(refreshExchange());
     store.dispatch(refreshOrders());
     const userAccount = store.getState().web3Connect.userAccount;
+    store.dispatch(processNewTrade('NewOrder', userAccount, this));
```

Comment:
this only needs to be dispatched if userAccount === maker, otherwise we display the trades of other accounts?